lib/vtls/schannel.c

```diff
@@ -2472,6 +2459,32 @@ static void schannel_close(struct Curl_cfilter *cf, struct Curl_easy *data)
 
 static int schannel_init(void)
 {
+  bool wine = FALSE;
+  bool wine_has_alpn = FALSE;
+
+#ifndef CURL_WINDOWS_UWP
+  typedef const char *(APIENTRY *WINE_GET_VERSION_FN)(void);
+  /* GetModuleHandle() not available for UWP.
+     Assume no WINE because WINE has no UWP support. */
+  WINE_GET_VERSION_FN s_p_wine_get_version =
+    CURLX_FUNCTION_CAST(WINE_GET_VERSION_FN,
+                        (GetProcAddress(GetModuleHandle(TEXT("ntdll")),
+                                        "wine_get_version")));
+  wine = !!s_p_wine_get_version;
+  if(wine) {
+    const char *wine_version = s_p_wine_get_version(); /* e.g. "6.0.2" */
+    /* Assume ALPN support with WINE 6.0.0 or upper */
+    wine_has_alpn = wine_version && atoi(wine_version) >= 6;
+  }
+#endif
+  /* ALPN is only supported on Windows 8.1 / Server 2012 R2 and above.
+     It is also supported under WINE 5.6+ with GnuTLS 3.2.0+ */
```

Comment:
This sentence should probably be clarified or removed, like reading this I'm wondering why if WINE ALPN is supported since 5.6 that we only enable since 6.0

Reply:
6 is the one I could test with. 5.6 is when support appeared in the sources where I looked, which may or may not mean actually working support. No release notes mention "ALPN", though the search on the WINE page seems suspiciously broken. Perhaps someone has better information.

Using the major version also simplifies parsing the version string.

I believe 6 is better than nothing and if someone reports it working earlier, it can be adjusted.

Reply:
What would be a better comment here?

Reply:
Adjusted the comment.